### ISSUE TITLE
ux-iconbar doesn't propagate data changes to ux-iconbaritem

### ISSUE THREAD

**Issue by ScottMaclure** [ractive-foundation/ractive-foundation#77]
Issue: when changing data on ux-iconbar, that's fine, but it doesn't pass data down to the child components.

Load http://localhost:9080/data.html and run in console:

``` javascript
window.RactiveF.widgets[1].findAllChildComponents('ux-iconbar')[0].set({
    "class": "label-right",
    "items": [
        {
            "href": "/new/path/to/blah",
            "src": "images/fi-info.svg",
            "label": "Homezzzzz"
        },
        {
            "href": "/path/to/blah",
            "src": "images/fi-info.svg",
            "label": "Bookmarkkkzkzkz"
        },
        {
            "src": "images/fi-info.svg",
            "label": "Infooijeoie",
            "class": "disabled"
        }
    ],
    "isDataModel": true
});
```

Note: Also needs to work for markup-driven components.

**Comment by ScottMaclure** [comment]
Refactored the solution tonight.

The parent component has two changes:
1. Added code only to deal with markup-driven component setup (so if you don't care about that, these changes don't matter).
2. Used a data function `getItemData` for passing data down to child (isolating data).

I've augmented `Ractive.prototype` with an `onrender` method, which observes for `datamodel` changes on that component, and then lifts the data into the root-level data.

The child will now update itself accordingly, as expected.

Edit: @pvshum @steven-bruce-au @bobodevil will find this refactor of interest.
